File: src/main/java/com/ultreon/mods/screenshotmanager/common/DoubleSize.java
package com.ultreon.mods.screenshotmanager.common;

import lombok.AllArgsConstructor;

import java.util.Objects;

@AllArgsConstructor
public class DoubleSize {
    public double width;
    public double height;

    public double getWidth() {
        return width;
    }

    public void setWidth(double width) {
        this.width = width;
    }

    public double getHeight() {
        return height;
    }

    public void setHeight(double height) {
        this.height = height;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        DoubleSize that = (DoubleSize) o;
        return Double.compare(that.width, width) == 0 && Double.compare(that.height, height) == 0;
    }

    @Override
    public int hashCode() {
        return Objects.hash(width, height);
    }
}

File: src/main/java/com/ultreon/mods/screenshotmanager/util/DoubleSize.kt
package com.ultreon.mods.screenshotmanager.util

import java.util.*

class DoubleSize(var width: Double, var height: Double) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as DoubleSize
        return that.width.compareTo(width) == 0 && that.height.compareTo(height) == 0
    }

    override fun hashCode(): Int {
        return Objects.hash(width, height)
    }
}